libraries/base/Data/Semigroup.hs

```diff
@@ -18,20 +18,33 @@
 -- Maintainer  :  libraries@haskell.org
 -- Stability   :  provisional
 -- Portability :  portable
+-- 
+-- A type @a@ is a 'Semigroup' if it provides an associative function ('<>')
+-- that lets you combine any two values of type @a@ into one. 
 --
--- In mathematics, a semigroup is an algebraic structure consisting of a
--- set together with an associative binary operation. A semigroup
--- generalizes a monoid in that there might not exist an identity
--- element. It also (originally) generalized a group (a monoid with all
--- inverses) to a type where every element did not have to have an inverse,
--- thus the name semigroup.
+-- Intuitively, this gives rise to an abstraction that allows for the
+-- combination of multiple values into a single one. The instance of 
+-- the 'Semigroup' determines the way in which the values are to be 
+-- combined.
 --
--- The use of @(\<\>)@ in this module conflicts with an operator with the same
--- name that is being exported by Data.Monoid. However, this package
--- re-exports (most of) the contents of Data.Monoid, so to use semigroups
--- and monoids in the same package just
```

Comment:
This paragraph seems like a useful thing to keep around.

Reply:
Yeah I agree.

Reply:
fair, it's back in ;)